code-name-taran/src/com/spaghettisoft/component/ninemenmorris/BoardState.java

```diff
@@ -0,0 +1,344 @@
+
+                    foundMills = true;
+                    drawGame();
+                    millsCounter++;
+                    System.out.println("Mill found at " + mills[i][0] + mills[i][1] + mills[i][2] + " !");
+                    // System.out.println(foundMills);
+                }
+            }
+        }
+        if (foundMills) { // Calls "mills" times deletePool();
+            for (int i = 0; i < millsCounter; i++) {
+                System.out.println("Enter an oponent place to delete: ");
+                String placeToDelete = StaticObjects.scanner.nextLine().toUpperCase();
+                // System.out.println(placeToDelete);
+                deletePool(placeToDelete);
+            }
+        }
+
+        return foundMills; // Redundant
+    }
+
+    private static BoardTile getBoardTileByPlace(String place) {
+        for (int i = 0; i < boardArrays.length; i++) {
+            for (int j = 0; j < boardArrays[i].length; j++) {
+                if (boardArrays[i][j].getBackground().equals(place)) {
+                    return boardArrays[i][j];
+                }
+            }
+        }
+        throw new IllegalArgumentException("Illegal place");
+    }
+
+    private static void checkForEndOfGame(String deletedPoolType) {
+
+        int poolCounterOponent = 0;
+
+        // BoardTile tileLastMove = getBoardTileByPlace(placeToDelete);
+
+        if (deletedPoolType.equals(BoardTile.WHITE)) {
+
+            for (int i = 0; i < boardArrays.length; i++) {
+                for (int j = 0; j < boardArrays[i].length; j++) {
+
+                    if (boardArrays[i][j].getPoolType().equals(BoardTile.WHITE)) {
+                        poolCounterOponent++;
+                    }
+                }
+            }
+            // System.out.println("White pools" + poolCounterOponent);
+            if ((poolCounterOponent < 3) && (placedPools == 18)) {
+                gameNotEnded = false;
+                System.out
+                        .println("Black player wins !!! White player has only " + poolCounterOponent + " pools left!");
+            }
+        }
+
+        else if (deletedPoolType.equals(BoardTile.BLACK)) {
+            for (int i = 0; i < boardArrays.length; i++) {
+                for (int j = 0; j < boardArrays[i].length; j++) {
+
+                    if (boardArrays[i][j].getPoolType().equals(BoardTile.BLACK)) {
+                        poolCounterOponent++;
+                    }
+                }
+            }
+            // System.out.println("Black pools " + poolCounterOponent);
+            if ((poolCounterOponent < 3) && (placedPools == 18)) {
+                gameNotEnded = false;
+                System.out
+                        .println("White player wins !!! Black player has only " + poolCounterOponent + " pools left!");
+            }
+        } else {
+            System.out.println("EMPTY pools" + poolCounterOponent);
+            System.out.println("emty pull type");
+        }
+    }

```

Comment:
It looks to me that the duplication of code here can be removed by making the method looks like this:
 + private static boolean checkForEndOfGame(String deletedPoolType) {
 +
 +        int poolCounterOponent = 0;
 +        for (int i = 0; i < boardArrays.length; i++) {
 +                for (int j = 0; j < boardArrays[i].length; j++) {
 +
 +                    if (boardArrays[i][j].getPoolType().equals(deletedPoolType)) {
 +                        poolCounterOponent++;
 +                    }
 +                }
 +            }
 +            // System.out.println("Black pools " + poolCounterOponent);
 +            if ((poolCounterOponent < 3) && (placedPools == 18)) {
 +                return false;
 +            }
 +           return true;
 + }

Changing the field and printing the message can be done outside of this method :P